povertymapping/hdx.py

```diff
@@ -210,4 +209,32 @@ def get_hdx_file(
 
     logger.info(f"HDX Data: Successfully downloaded and cached for {region} at {zipfile_path}!")
 
-    return unzipped_hdxfile
\ No newline at end of file
+    return unzipped_hdxfile
+
+
+def generate_hdx_features(
+    aoi: pd.DataFrame,
+    region: str,
+    extra_args: dict = dict(nodata=np.nan),
+) -> pd.DataFrame:
+    """
+    Append HRSL population from HDX to existing DataFrame
+
+    Args:
+        aoi (pandas DataFrame): The input AOI dataframe.
+        region (str): Country/territory ISO3 region name (see iso3.get_region_name()).
+        extra_args (dict, optional): Additional arguments to raster zonal stats (see geowrangler.raster_zonal_stats). Defaults to dict(nodata=np.nan).
+    
+    Returns:
+        aoi (pd.DataFrame): The AOI dataframe with its new features.
+    """
+    hdx_pop_file = get_hdx_file(region)
+
+    aoi = rzs.create_raster_zonal_stats(
+        aoi,
+        hdx_pop_file,
+        aggregation=dict(column="population", output="pop_count", func="sum"),
```

Comment:
not urgent: could we also include `mean` for population density? I think we can pass a list to `func` and `output` 

We can wait a bit to implement this before we merge the PR

Reply:
since this function takes in arbitrary AOIs (not necessarily quadkeys), might be "best practice" to include density by default also

Reply:
Alright! To clarify, taking the `mean` in this case is already the population density, right?

Reply:
yes! in that case might be clear also to call it `population_density` and `total_population` vs `pop_mean` and `pop_sum` etc haha

Reply:
From our discussions, retained only `population_density` as the new feature.